metron-platform/metron-enrichment/metron-enrichment-common/src/main/java/org/apache/metron/enrichment/cache/ObjectCache.java

```diff
@@ -0,0 +1,118 @@
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class ObjectCache {
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  protected LoadingCache<String, Object> cache;
+  private static ReadWriteLock lock = new ReentrantReadWriteLock();
+  protected LoadingCache<String, Object> getCache() {
+    return cache;
+  }
+
+  public class Loader implements CacheLoader<String, Object> {
+    FileSystem fs;
+    ObjectCacheConfig objectCacheConfig;
+
+    public Loader(Configuration hadoopConfig, ObjectCacheConfig objectCacheConfig) throws IOException {
+      this.fs = FileSystem.get(hadoopConfig);
+      this.objectCacheConfig = objectCacheConfig;
+    }
+
+    @Override
+    public Object load(String s) throws Exception {
+      LOG.debug("Loading object from path '{}'", s);
+      if (StringUtils.isEmpty(s)) {
+        throw new IllegalArgumentException("Path cannot be empty");
+      }
+      Object object = null;
+      Path p = new Path(s);
+      if (fs.exists(p)) {
+        if (fs.getFileStatus(p).getLen() <= objectCacheConfig.getMaxFileSize()) {
+          try (InputStream is = new BufferedInputStream(fs.open(p))) {
+            byte[] serialized = IOUtils.toByteArray(is);
+            if (serialized.length > 0) {
+              object = SerDeUtils.fromBytes(serialized, Object.class);
+            }
+          }
+        } else {
+          throw new IllegalArgumentException(String.format("File at path '%s' is larger than the configured max file size of %s", p, objectCacheConfig.getMaxFileSize()));
+        }
+      } else {
+        throw new IllegalArgumentException(String.format("Path '%s' could not be found in HDFS", s));
+      }
+      return object;
+    }
+  }
+
+  public Object get(String path) {
+    return cache.get(path);
+  }
+
+  public void initialize(ObjectCacheConfig config) {
+    try {
+      lock.writeLock().lock();
+
+      cache = setupCache(config);
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to initialize: " + e.getMessage(), e);
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  public boolean isInitialized() {
+    try {
+      lock.readLock().lock();
+      return cache != null;
+    } finally {
+      lock.readLock().unlock();
+    }
+  }
+
+  protected LoadingCache<String, Object> setupCache(ObjectCacheConfig config) throws IOException {
+    LOG.info("Building ObjectCache with {}", config);
+    return Caffeine.newBuilder().maximumSize(config.getCacheSize())
+            .expireAfterWrite(config.getCacheExpiration(), config.getTimeUnit())
+            .removalListener((path, value, removalCause) -> {
+              LOG.debug("Object retrieved from path '{}' was removed with cause {}", path, removalCause);
+            })
+            .build(new Loader(new Configuration(), config));
```

Comment:
If there's one place to provide an opportunity for dependency injection, I think the `Loader` implementation is it.